crates/bevy_entropy/src/lib.rs

```diff
@@ -0,0 +1,139 @@
+use bevy_app::{App, Plugin};
+use bevy_utils::tracing::{debug, trace};
+use rand::{rngs::StdRng, RngCore, SeedableRng};
+
+pub mod prelude {
+    #[doc(hidden)]
+    pub use crate::Entropy;
+}
+
+/// Provides a source of entropy.
+/// This enables deterministic random number generation.
+///
+/// See <https://github.com/bevyengine/bevy/discussions/2480> for issues
+/// to be mindful of if you desire complete determinism.
+#[derive(Default)]
+pub struct EntropyPlugin;
+
+impl Plugin for EntropyPlugin {
+    fn build(&self, app: &mut App) {
+        if !app.world.contains_resource::<Entropy>() {
+            trace!("Creating entropy");
+            app.init_resource::<Entropy>();
+        }
+    }
+}
+
+/// A resource that provides entropy.
+pub struct Entropy(StdRng);
+
+impl Default for Entropy {
+    /// The default entropy source is non-deterministic and seeded from the operating system.
+    /// For a deterministic source, use [`Entropy::from`].
+    fn default() -> Self {
+        debug!("Entropy created via the operating system");
+        let rng = StdRng::from_entropy();
+        Entropy(rng)
+    }
+}
+
+impl Entropy {
+    /// Create a deterministic source of entropy. All random number generators
+    /// later seeded from an [`Entropy`] created this way will be deterministic.
+    /// If determinism is not required, use [`Entropy::default`].
+    pub fn from(seed: [u8; 32]) -> Self {
+        debug!("Entropy created via seed: {:?} ", seed);
+        let rng = StdRng::from_seed(seed);
```

Comment:
`StdRng` does not have cross-platform or cross-version reproducibility. Certainly not desirable if we want determinism. We may want to use the underlying RNG directly, which is ChaCha12 at the moment.

Reply:
Good idea.